### PR TITLE
support uuid in event payloads

### PR DESCRIPTION
**Pull request by yakkomajuri**
## Changes

*Please describe.*  
*If this affects the frontend, include screenshots.*  

Tiny change but it's actually **massive** (in a good way, not a breaking way).

This makes it so that people can send events to us with a UUID specified, which means they can send an event 3x but we'll eventually collapse it in ClickHouse.

Think about why this is big: plugins exporting historical events for migration purposes (e.g. Migrator 3000) can be defensive and follow an approach of "export at least once", which guarantees delivery but could end up sending duplicates.

This can also be applied to any pipeline sending data to us, be it one of our own libraries or a CDP like Segment or RudderStack.

## How did you test this code?

*Please describe.*

Added a test.

### CODE REVIEW

File: posthog/api/capture.py [hunk 1]
```diff
@@ -205,6 +205,10 @@ def get_event(request):
         if not distinct_id:
             continue
 
+        payload_uuid = event.get("uuid", None)
+        if payload_uuid and UUIDT.is_valid_uuid(payload_uuid):
+            event_uuid = UUIDT(uuid_str=payload_uuid)
```
Comment (tiina303) [inline posthog/api/capture.py:210]: Let's respond with `invalid payload` instead if the uuid is defined but isn't valid as us then overriding it with a random one would be unexpected and difficult to debug in the future.

**Review by tiina303 — APPROVED** [review]
address the comment before merging

File: posthog/api/capture.py (hunk 1 above)
Comment (yakkomajuri) [reply posthog/api/capture.py:210]: Hmm this is a good point, but it's problematic here after the refactor.

People send us events in a batch, and we don't have an amazing story yet for how to handle invalid payloads within the batch. Sure we could respond with something like 'ingested these and not these' but that needs some careful consideration. it also would prevent us from improving latency down the line by dumping events into another Kafka queue immediately as we get them.

Currently invalid payloads get ignored, and in this case it's better to default to a random ID than it is to ignore imo.

Maybe a thing to do here is add metrics. Something for the future would also be a dashboard with an overview of ingestion.

Comment (yakkomajuri) [reply posthog/api/capture.py:210]: Anchor links seem broken but check out "Invalid events" here: https://posthog.com/docs/api/post-only-endpoints

Comment (tiina303) [reply posthog/api/capture.py:210]: maybe we can add a log then (Sentry), so we at least know how often this is happening (so we know to prioritise)?

Comment (yakkomajuri) [reply posthog/api/capture.py:210]: Will add a statsd metric

**Pull request merged**